railgun/test/integration/plugin_test.go

```diff
@@ -0,0 +1,154 @@
+	assert.NoError(t, err)
+
+	defer func() {
+		t.Logf("ensuring that Ingress %s is cleaned up", ingress.Name)
+		if err := cluster.Client().NetworkingV1().Ingresses(corev1.NamespaceDefault).Delete(ctx, ingress.Name, metav1.DeleteOptions{}); err != nil {
+			if !errors.IsNotFound(err) {
+				require.NoError(t, err)
+			}
+		}
+	}()
+
+	t.Logf("waiting for routes from Ingress %s to be operational", ingress.Name)
+	p := proxyReady()
+	assert.Eventually(t, func() bool {
+		resp, err := httpc.Get(fmt.Sprintf("%s/httpbin", p.ProxyURL.String()))
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", p.ProxyURL.String(), err)
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusOK {
+			// now that the ingress backend is routable, make sure the contents we're getting back are what we expect
+			// Expected: "<title>httpbin.org</title>"
+			b := new(bytes.Buffer)
+			b.ReadFrom(resp.Body)
+			return strings.Contains(b.String(), "<title>httpbin.org</title>")
+		}
+		return false
+	}, ingressWait, waitTick)
+
+	kongplugin := &kongv1.KongPlugin{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: corev1.NamespaceDefault,
+			Name:      "teapot",
+		},
+		PluginName: "request-termination",
+		Config: apiextensionsv1.JSON{
+			Raw: []byte(`{"status_code": 418}`),
+		},
+	}
+	kongclusterplugin := &kongv1.KongClusterPlugin{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "legal",
+		},
+		PluginName: "request-termination",
+		Config: apiextensionsv1.JSON{
+			Raw: []byte(`{"status_code": 451}`),
+		},
+	}
+	c, err := clientset.NewForConfig(cluster.Config())
+	assert.NoError(t, err)
+	kongplugin, err = c.ConfigurationV1().KongPlugins(corev1.NamespaceDefault).Create(ctx, kongplugin, metav1.CreateOptions{})
+	assert.NoError(t, err)
+	kongclusterplugin, err = c.ConfigurationV1().KongClusterPlugins(corev1.NamespaceDefault).Create(ctx, kongclusterplugin, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	ingress, err = cluster.Client().NetworkingV1().Ingresses(corev1.NamespaceDefault).Get(ctx, ingress.Name, metav1.GetOptions{})
+	assert.NoError(t, err)
+
+	t.Logf("updating Ingress %s to use plugin %s", ingress.Name, kongplugin.Name)
+	ingress.ObjectMeta.Annotations[annotations.AnnotationPrefix+annotations.PluginsKey] = kongplugin.Name
+	ingress, err = cluster.Client().NetworkingV1().Ingresses(corev1.NamespaceDefault).Update(ctx, ingress, metav1.UpdateOptions{})
+	assert.NoError(t, err)
+	assert.Eventually(t, func() bool {
+		resp, err := httpc.Get(fmt.Sprintf("%s/httpbin", p.ProxyURL.String()))
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", p.ProxyURL.String(), err)
+			return false
+		}
+		defer resp.Body.Close()
+		return resp.StatusCode == http.StatusTeapot
+	}, ingressWait, waitTick)
+
+	// TODO as of yet, this does not work. The controller does not detect the plugin and cannot apply it.
```

Comment:
Could you please create an issue in the beta milestone for this?

Reply:
https://github.com/Kong/kubernetes-ingress-controller/issues/1361